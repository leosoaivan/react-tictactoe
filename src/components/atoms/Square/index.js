import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import LetterX from '../../../images/Letter-X.png';
import LetterO from '../../../images/Letter-O.png';

const Root = styled.div`
  background: white;
  height: 100%;
  width: 100%;
`;

const Symbol = styled.img`
  box-sizing: border-box;
  object-fit: contain;
  max-height: 100%;
  max-width: 100%;
  padding: 5px;
`;

const Square = ({ onClick, value }) => {
  const symbol = () => {
    if (value === 'X') {
      return <Symbol src={LetterX} alt="" />;
    }

    if (value === 'O') {
      return <Symbol src={LetterO} alt="" />;
    }

    return null;
  };

  return (
    <Root onClick={onClick}>
      {symbol()}
    </Root>
  );
};

Square.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string,
};

Square.defaultProps = {
  onClick: null,
  value: null,
};

export default Square;
