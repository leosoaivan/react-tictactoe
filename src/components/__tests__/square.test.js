import React from 'react';
import { shallow } from 'enzyme';
import Square from '../square';
import {createSerializer} from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Square', () => {
  it('Renders the Square component', () => {
    const wrapper = shallow(
      <Square></Square>
    )

    expect(wrapper).toMatchSnapshot();
  })
})