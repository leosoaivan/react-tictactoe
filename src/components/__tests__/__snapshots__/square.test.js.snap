// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Renders the Square component 1`] = `
<styled.div
  onClick={[Function]}
/>
`;
